compute_tools/src/vm_monitor/cgroup.rs

```diff
@@ -0,0 +1,777 @@
+use std::{
+    fmt::{Debug, Display},
+    fs,
+    pin::Pin,
+    sync::atomic::{AtomicU64, Ordering},
+    task::{self, Poll},
+};
+
+use anyhow::{anyhow, Context};
+use async_std::channel::{Receiver, Sender};
```

Comment:
Should instead use tokio::sync channels. If these are used because they don't (intentionally) implement Stream, they can be found from `tokio-stream`.

Reply:
Thanks for the feedback. I did some investigation into channels at the beginning of writing the monitor and eventually settled on `async-std`'s for a couple reasons. Firstly, the channel need to be at the very least SPMC (so I guess MPMC is fine too). Here are some of the choices I reviewed:
* `tokio::sync::broadcast`: each message is broadcast to _every_receiver_, while I want more of a multi-consumer queue like system where receiving _removes_ items from the queue so they are only processed once.
* `tokio::sync::watch`: you can see if the message has changed, and you can see the latest message, but there is no way of receiving the latest message _and_ marking it as seen so that other threads do not read it. Once again, not satisfying the `remove-from-queue` behaviour I want.
* `crossbeam-channel`: works as a queue way I want, but I cannot merge multiple channels together because `Receiver`s do not implement `Stream`.
* `async-std`: works as a queue the way I want, and `Receiver`s also implement stream. :heavy_check_mark:.

The reason I need the `Stream/StreamExt` traits is for merging/peeking.

I may be missing something - please correct me if so. I agree that we would not like to be stuck on `futures = ^0.3`.

Reply:
It seems the multiple consumers aspect of async-channel is equal to "one consumer of many gets the message", not "all consumers get the message."

Is that what is wanted? Wrapping tokio::sync::Receiver in a mutex will do the same.

What part of code relies on the multiple consumers? Is it okay if only one of the consumers gets all of the messages?